.github/workflows/checks.yaml

```diff
@@ -0,0 +1,38 @@
+name: Checks
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+    branches: [master]
+jobs:
+  count_commits:
+    name: Check that there is only one commit
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+    - name: Dump GitHub context (for debugging)
+      env:
+        GITHUB_CONTEXT: ${{ toJSON(github) }}
+      run: |
+        echo "GITHUB_CONTEXT: $GITHUB_CONTEXT"
+    - name: Check number of commits
+      shell: bash --norc --noprofile {0}
+      run: |
+        cat <<EOF | egrep -qsi '^disable-check:.*\<commit-count\>'
```

Comment:
This is quite non-obvious. I suggest adding comments that describe what this is doing. It would also be good to have a top-level comment/header documenting generally what this workflow is doing and how to properly use it.

I think this is about a way to disable this check in a comment, but it is unclear.

Reply:
Also, maybe it is better to disable this in a comment rather than updating the PR body? We probably don't want people to commit this metadata to the git log, which I am sure will happen if this is part of the PR body given how these two are connected.

Reply:
Added a comment.

Reply:
> In fact, I see that this is already the case in this PR's commit.

Uhm, where?

Reply:
I agree with @erimatnor that the message to disable check should not be part of the message of this commit. However, in general it will not be part of a commit message, but only in PR, since in the case of several commits PR post doesn't match commit messages.

Reply:
> I agree with @erimatnor that the message to disable check should not be part of the message of this commit.

It is part of the message, since it is needed for the description of the commit.

Reply:
> However, in general it will not be part of a commit message, but only in PR, since in the case of several commits a PR post doesn't match the commit messages.

Correct, that is why it only checks the PR message, not the commits.

Reply:
> Also, maybe it is better to disable this by adding a comment rather than updating the PR body? We probably don't want people to commit this metadata to the git log, which I am sure will happen if this is part of the PR body given how these two are connected.

If you have multiple commits, the default is to not use either of them, so it is not likely that anybody deliberately adds several commits and have the header/trailer line in one of the commits.

Reply:
My point was that having the check on the PR body is a bad idea since it will lead to people including the metadata in the commit log. As I pointed out, this already happened in this PR and is now merged: https://github.com/timescale/timescaledb/commit/17574807b098a389ec2ac31f74e530abb1988cd5

So the very thing I warned about happened in this PR.

Reply:
No, that is not metadata, this is an example of the row that you should add.